### PR TITLE
Remove docs pip extra from setup.py

### LINKED ISSUE #2524: Distributions must not conatin direct dependency references to github
```
python3 -m twine upload dist/* --verbose
...
400 Invalid value for requires_dist. Error: Can't have direct dependency: "jsonschema2rst @ git+git://github.com/inspirehep/jsonschema2rst.git@d211d9709046431a6b6a4594c97c22d8713d854b ; extra == 'deploy'
```

### PR DESCRIPTION
**Pull request by KPrasch**
**Type of PR:**
- [X] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [X] 1
- [ ] 2
- [ ] 3

**What this does:**

Removes the docs pip extra to allow prevent distributing `jsonschema2rst`

**Issues fixed/closed:**

Closes #2524

### CODE REVIEW

File: setup.py
Comment (derekpierre) [inline setup.py:142]: Can we keep the 'docs' directive, and just not include it as part of the 'dev' and 'deploy' directives?

Comment (KPrasch) [reply setup.py:142]: Unfortunately not, since the dependencies defined in the pip extra are included as part of the distribution's metadata causing twine to fail as shown in #2524

Comment (KPrasch) [reply setup.py:142]: I too am am big fan of pip extras, I think the longer term solution here is to either fork `jsonschema2rst`, wait for the maintainer to merge https://github.com/inspirehep/jsonschema2rst/pull/8, or create another custom solution entirely.

Comment (derekpierre) [reply setup.py:142]: Ok - perhaps we can revisit once `jsonschema2rst` is updated.

**Review by derekpierre — APPROVED** [review]
🎸

**Pull request merged**
